docs/2.8.0/docs/daml-finance/reference/patterns.rst

```diff
@@ -51,6 +51,35 @@ could in principle stay untouched.
 For an example where the Factory pattern is used, check out the
 :doc:`Holdings tutorial <../tutorials/getting-started/holdings>`.
 
+.. _reference-pattern:
+
+Reference pattern
+-----------------
+
+The Reference pattern is used to leverage the functionalities of :ref:`Contract Keys <contractkeys>`
+when working with interfaces.
+
+We want for instance to use an
+:ref:`InstrumentKey <constr-daml-finance-interface-types-common-types-instrumentkey-32970>` to
+identify instruments across a number of implementing templates.
+
+To do that, we define a `Reference` template that
+ - is keyed by the `InstrumentKey`
+ - contains the contract id of the target instrument template (e.g. a fixed rate bond)
+
+We can then fetch an interface "by key" by 
+ - fetching the `Reference` template by key (``fetchByKey``)
+ - reading and ``fetch``-ing the stored contract id
```

Comment:
Do we maybe want to explain where the interface comes in? We say `We can then fetch an interface "by key" by` but then only mention `Reference` and `contract id`.

Reply:
It is shown in the picture (the Reference template contains the `cid : ContractId Instrument`) but you are right that we don't specifically mention the interface in the wording